### PR TITLE
replace[Regexp]{One,All}(): allow non-const pattern and replacement arguments

### PR DESCRIPTION
**Pull request by rschu1ze**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Functions replaceOne(), replaceAll(), replaceRegexpOne() and replaceRegexpAll() can now be called with non-const pattern and replacement arguments

### CODE REVIEW

**Comment by rschu1ze** [comment]
Test failures are unrelated.

File: src/Functions/ReplaceStringImpl.h [hunk 1]
```diff
@@ -16,27 +21,33 @@ struct ReplaceStringTraits
         All
     };
 };
+
 /** Replace one or all occurencies of substring 'needle' to 'replacement'. 'needle' and 'replacement' are constants.
   */
```
Comment (CurtizJ) [inline src/Functions/ReplaceStringImpl.h:26]: Need to update comment.

File: src/Functions/ReplaceStringImpl.h [hunk 2]
```diff
@@ -85,34 +96,275 @@ struct ReplaceStringImpl
...
+            }
+
+            /// Copy suffix after last match
+            size_t bytes = (last_match == nullptr) ? (cur_haystack_end - cur_haystack_data + 1)
+                                                   : (cur_haystack_end - last_match - cur_needle_length + 1);
+            res_data.resize(res_data.size() + bytes);
+            memcpy(&res_data[res_offset], start_pos, bytes);
+            res_offset += bytes;
+
+            res_offsets[i] = res_offset;
```
Comment (CurtizJ) [inline src/Functions/ReplaceStringImpl.h:339]: Maybe this part of code can be unified between `vectorVectorConstant`, `vectorConstantVector` and `vectorVectorVector` like it's done in `RegexSearcher` with `processString` method?

File: src/Functions/ReplaceStringImpl.h (hunk 1 above)
Comment (rschu1ze) [reply src/Functions/ReplaceStringImpl.h:26]: Done

File: src/Functions/ReplaceStringImpl.h (hunk 2 above)
Comment (rschu1ze) [reply src/Functions/ReplaceStringImpl.h:339]: I'll give it a try

Comment (rschu1ze) [reply src/Functions/ReplaceStringImpl.h]: I tried an approach similar to `processString`, but the code does not really get better by that. Reason is that there is a lot less shared code here than in `processString`. What I did was to factorize the copy code, that saves a little bit at least.

**Comment by rschu1ze** [comment]
00002_log_and_exception_messages_formatting is flaky --> #48441

@CurtizJ Approve? Thanks.

**Review by CurtizJ — APPROVED** [review]

**Pull request merged**